packages/web/hooks/ui-config/use-add-concentrated-liquidity-config.ts

```diff
@@ -98,5 +101,33 @@ export function useAddConcentratedLiquidityConfig(
     config.poolId,
   ]);
 
-  return { config, addLiquidity };
+  const increaseLiquidity = useCallback(
+    (positionId: string) => {
+      return new Promise<void>(async (resolve, reject) => {
```

Comment:
the structure of the promise executor is a little confusing to me. I might be missing something, but would something like this work?

```
const increaseLiquidity = useCallback(
    (positionId: string): Promise<void> => {
      const amount0 = config.baseDepositAmountIn.getAmountPrimitive().amount;
      const amount1 = config.quoteDepositAmountIn.getAmountPrimitive().amount;

      return account.osmosis
        .sendAddToConcentratedLiquidityPositionMsg(
          positionId,
          amount0,
          amount1
        )
        .then((tx) => {
          if (tx.code) throw new Error(tx.log);
        })
        .catch((e: any) => {
          throw new Error(e.message);
        });
    },
    [config.baseDepositAmountIn, config.quoteDepositAmountIn, account.osmosis]
  );
  ```

Reply:
Yeah I think this API could be improved, it originates from keplr-wallet libs. I believe the only place the tx contents are returned is in the onFullfilled callback passed into the send* funtion.

Reply:
I believe the `send*` function(s) (ours) which calls `sendMsgs` (keplr-wallet libs) on cosmos base account resolves to `undefined` when the tx is _broadcast_ (i.e. they click sign). However, it sets up a [websocket listener](https://github.com/chainapsis/keplr-wallet/blob/7a4db487b74170d968ecc2c7979d39579251277c/packages/stores/src/account/cosmos.ts#L441) to receive push updates as the tx gets processed, then returns the committed tx object in the `onFullfilled` callback.

See def for `sendMsgs`:
https://github.com/chainapsis/keplr-wallet/blob/7a4db487b74170d968ecc2c7979d39579251277c/packages/stores/src/account/cosmos.ts#L363

With @JoseRFelix 's work to pull send logic into our code, we can make changes to this if we wish. We discussed this a bit but I believe Jose chose to stick with this API (resolve top level fn on broadcast, pass callback for fullfill).